README.md

```diff
@@ -2,54 +2,181 @@
+
+```bash
+export KUBECOMPOSE_NAMESPACE=""
+export KUBECOMPOSE_ENVID=""
 ```
-This writes to the directory `test/output` the created Kubernetes resources.
 
-To clean up after the test:
+## Examples
+
+kube-compose loads pod and services definitions implicitly defined in a docker compose file, and creates them in a target namespace via the following command:
+
+```bash
+kube-compose -e mybuildid up
 ```
-kubectl delete $(kubectl get all -lenv=test123 -oname)
+
+The target namespace and service account token are loaded from the context set in `~/.kube/config`. This means that k8s Client Tools kubectl commands can be used to configure kube-compose's target namespace and service account.
+
+If no `~/.kube/config` exists and kube-compose is run inside a pod in Kubernetes, the pod's namespace becomes the target namespace, and the service account used to create pods and services is the pod's service account.
+
+The namespace can be overridden via the `--namespace` option, for example: `kube-compose --namespace ci up`.
+
+### Foreground mode to view the logs of running pods
+
+```bash
+kube-compose --namespace default --env-id test123 up
+
+kube-compose --namespace default --env-id test123 down
+```
+
+```bash
+kube-compose up -n default -e test123
+
+kube-compose down -n default -e test123
+
+```
+
+If environment variables are already set.
+
+```bash
+kube-compose up
+
+kube-compose down
+```
+
+Start individual services defined in docker-compose.yml
+
+```bash
+kube-compose up service-1
+
+kube-compose up service-1 service-2
+```
+
+### Detach mode
+
+```bash
+kube-compose --namespace default --env-id test123 up --detach
+```
+
+```bash
+kube-compose up -n default -e test123 -d
+```
+
+If environment variables are already set.
+
+```bash
+kube-compose up -d
 ```
+
+## Why another tool
+
+Although [kompose](https://github.com/kubernetes/kompose) can already convert docker compose files into Kubernetes resources. The main differences between kube-compose and Kompose are:
+
+1. kube-compose generates Kubernetes resource names and selectors that are unique for each build to support shared namespaces and scaling to many concurrent CI environments.
+
+1. kube-compose creates pods with `restartPolicy: Never` instead of deployments, so that failed pods can be inspected, no logs are lost due to pod restarts, and Kubernetes cluster resources are used more efficiently.
+
+1. kube-compose allows startup dependencies to be specified by respecting [docker compose](https://docs.docker.com/compose/compose-file/compose-file-v2#depends_on)'s `depends_on` field.
+
+1. kube-compose currently depends on the docker daemon to pull Docker images and extract their healthcheck.
+
+## Advanced Usage
+
+If you require that an application is not started until one of its dependencies is healthy, you can add `condition: service_healthy` to the `depends_on`, and give the dependency a [Docker healthchecks](https://docs.docker.com/engine/reference/builder#healthcheck).
```

Comment:
..."until some or all of its dependencies is healthy"...

"Docker healthcheck", not plural.

Reply:
fixed.